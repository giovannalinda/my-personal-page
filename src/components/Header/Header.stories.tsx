import { Meta, Story } from '@storybook/react'
import { Header } from './Header'

export default {
  title: 'Components/Header',
  component: Header,
} as Meta

export const Default: Story = () => <Header />
