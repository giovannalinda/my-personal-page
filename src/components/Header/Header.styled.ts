import styled from 'styled-components'
import { theme } from '~/styles'

export const Header = styled.header`
  width: 100%;
  position: fixed;
  background: ${theme.colors.black};
  color: ${theme.colors.white};
  border-bottom: 1px solid ${theme.colors.gray800};
  z-index: 3;
`

export const List = styled.ul`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: ${theme.space[5]} 10rem;

  svg {
    color: ${theme.colors.white};
  }
`

export const ListItem = styled.li`
  display: flex;
  align-items: center;

  h1,
  p {
    font-size: ${theme.font.sizes.paragraph};
  }

  p {
    padding-left: ${theme.space[0]};
  }

  small {
    font-size: ${theme.font.sizes.paragraph};
  }
`

export const OpenSidebarButton = styled.button`
  background: none;
`
