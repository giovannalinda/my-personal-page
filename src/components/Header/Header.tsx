import { AnimatePresence } from 'framer-motion'
import { useState } from 'react'
import { AiOutlineAlignRight } from 'react-icons/ai'
import { Sidebar } from '../Sidebar/Sidebar'

import * as S from './Header.styled'

export function Header() {
  const [isOpen, setIsOpen] = useState(false)

  function handleOpenSidebar() {
    setIsOpen(true)
  }

  function handleCloseSidebar() {
    setIsOpen(false)
  }
  return (
    <S.Header>
      <S.List>
        <S.ListItem>
          <h1>Giovanna</h1>
          <p>Souza</p>
        </S.ListItem>
        <S.ListItem>
          <small>PT EN</small>
        </S.ListItem>
        <S.ListItem>
          <S.OpenSidebarButton>
            <AiOutlineAlignRight size={24} onClick={handleOpenSidebar} />
          </S.OpenSidebarButton>

          <AnimatePresence>
            {isOpen && <Sidebar onClose={handleCloseSidebar} />}
          </AnimatePresence>
        </S.ListItem>
      </S.List>
    </S.Header>
  )
}
