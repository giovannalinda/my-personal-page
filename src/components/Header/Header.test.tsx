import { render } from '~/__helpers__/app-tests'
import { Header } from './Header'

describe('Header', () => {
  it('matches snapshot', () => {
    const { container } = render(<Header />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
