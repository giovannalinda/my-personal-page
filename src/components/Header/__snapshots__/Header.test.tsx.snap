// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  class="sc-papXJ ObRRU"
>
  <ul
    class="sc-jqUVSM kmzZuc"
  >
    <li
      class="sc-kDDrLX irZuR"
    >
      <h1>
        Giovanna
      </h1>
      <p>
        Souza
      </p>
    </li>
    <li
      class="sc-kDDrLX irZuR"
    >
      <small>
        PT EN
      </small>
    </li>
    <li
      class="sc-kDDrLX irZuR"
    >
      <button
        class="sc-iqcoie haNNNk"
      >
        <svg
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M904 158H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 424H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 212H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0-424H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </button>
    </li>
  </ul>
</header>
`;
